system-parachains/bridge-hubs/bridge-hub-kusama/src/lib.rs

```diff
@@ -460,7 +493,9 @@ construct_runtime!(
 		XcmpQueue: cumulus_pallet_xcmp_queue = 30,
 		PolkadotXcm: pallet_xcm = 31,
 		CumulusXcm: cumulus_pallet_xcm = 32,
+		// Temporary to migrate the remaining DMP messages:
 		DmpQueue: cumulus_pallet_dmp_queue = 33,
+		MessageQueue: pallet_message_queue = 34,
```

Comment:
Thanks for adding the MessageQueue pallet, @bkontur! Can you please move this pallet to be at index 250? We rely on the MessageQueue pallet to be after the Snowbridge pallets. More details here: https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/lib.rs#L731

Reply:
This has [come up before](https://github.com/paritytech/polkadot-sdk/pull/1378#issuecomment-1733250552), there are really two indices here, the `= 34` index, which specifies the pallet index in call encoding, and the order of the variants in `construct_runtime`, which specifies the order of `on_initialize` running.

With Construct Runtime v2, I think anything that depends on this behavior needs to have tests to avoid things breaking. See e.g. [Rococo's construct runtime](https://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.6.0/polkadot/runtime/rococo/src/lib.rs#L1338) where MMR's index is higher than Session's but it is "before" it.

All this is to say, Message Queue could stay at 34 but move to the end of construct runtime. Personally this seems like a footgun and I'd like a single order instead of two orders, but I think in the Relay Chains all of the low-number indices are claimed so there's not a good way around it without a breaking change. I think anything that depends on orders like this should have clear documentation and tests to guarantee that things like underlying changes to the `construct_runtime` macro don't inadvertently break things.

Reply:
thank you @joepetrowski, so I will move it at the end with index 34.

@claravanstaden could you:
1. confirm that `MessageQueue: pallet_message_queue = 34,` moved to the end is ok?
2. create an issue to the `polkadot-sdk` for adding a test-case to the `bridge-hub-test-utils` ensuring this ordering/indexing feature works as expected? I would like to reuse that test-case here and also for Rococo<>Eth

Reply:
Sorry it wasn't clear, I actually meant the opposite. We should try to make the indices match the order that things are in `construct_runtime`. So 250 is technically fine, but it also leaves very little room for things that would need to come after later.

Reply:
@bkontur 

1. I think the MessageQueue pallet needs to be larger than the EthereumOutboundQueue pallet index for this to work correct.
2. I logged issue https://github.com/paritytech/polkadot-sdk/issues/3208.

Reply:
The index doesn't change the order of `on_initialize`. The order of `on_initialize` is determined based on the order of the pallet in `construct_runtime!`. There is also no way that you can get this order right now in the code.

Reply:
> There is also no way that you can get this order right now in the code.

I mean, there is.. the literal order of the lines of code in `construct_runtime!` 😆  hard codes the order in which hooks are called for the pallets.

But yes, the pallet index number does not matter here and there is no way to specify a number or some other ordering mechanism than the literal order of declaration in the macro.

Reply:
Yip, the order of the pallets does what we need. 🤷🏻‍♀️

Reply:
Yes but please also change the index to be greater than whatever it is after. What I meant in my first comment is that by not doing so, we are setting ourselves up for something to break when there is a change to the `construct_runtime` macro.

Reply:
FWIW, in [CR v2](https://github.com/paritytech/polkadot-sdk/pull/1378), I am planning to make the order of hooks as per the pallet index, but support an attribute `legacy_ordering` that would follow the current approach. This would ensure that the older ones continue to work but the newer chains use the correct paradigm.

Reply:
Yeah but let's just order them both on the chains where we can, like here. I know there are some places on the Relay it's not possible, but I don't see a reason to add something here that we know we'll need to patch around with this `legacy_ordering`. We know about CRv2 so let's just make something that works.

Reply:
I hope I understood it well :D:
- if so, please, resolve this conversation
- if not, please, correct me

https://github.com/polkadot-fellows/runtimes/pull/137/commits/7fc05c1671f1f5b01ca26ad57c350c0dc87892be

moved to the end:
```
MessageQueue: pallet_message_queue = 175,
```

Reply:
Thanks @bkontur! Is it fine that the MessageQueue index is at `250` on Rococo and `175` here? I mean, it won't break anything but just wanted to note that it is different now. Is there a max pallet index?

Reply:
I would say that it is totally fine. If your pallet's common code (reused for Rococo and Polkadot) expects a hard-coded 250., then it is not fine and that code should be changed :)

Do you use MessageQueue and/or its index directly somewhere, e.g. as an encoded calls for Transact?

Reply:
I would not recommend 250, just because it means that in the future there are only 5 pallets that could possible go after it without a breaking change. Since (a) pallet order can be important, and (b) we have many subsystems that are compositions of several pallets (governance, bridges, staking, etc.) then putting it at 250 seems like we are setting ourselves up for problems later.

Reply:
> Do you use MessageQueue and/or its index directly somewhere, e.g. as an encoded calls for Transact?

No, we don't use it anywhere. So it should be fine.

Reply:
@joepetrowski agreed. The MessageQueue pallet index is just unfortunately already 250 on Rococo, but agreed 175 is better for Kusama and Polkadot.